8.gtex-interpret/scripts/nbconverted/4.visualize-signatures.r

```diff
@@ -440,25 +673,241 @@ head(full_heme_results_df, 3)
+      factor(final_results_df$algorithm,
+             levels = c("PCA", "ICA", "NMF", "DAE", "VAE"))
+  
+    final_results_df <- final_results_df %>% dplyr::arrange(z)
+
+    # Generate plot title
+    if (cell_type == "Neutrophil") {
+        gse <- "GSE103706"
+    } else {
+        gse <- "GSE24759"
+    }
+    plot_title <- paste(cell_type, "-", gse)
+
+    line_plot[[cell_type]] <- # Plot and save to file
+      ggplot(final_results_df,
+             aes(x = z,
+                 y = neg_log_p,
+                 color = algorithm,
+                 group = algorithm)) +
+      geom_point(size = 0.1) +
+      geom_smooth(lwd = 0.2,
+                  formula = y ~ x,
+                  method = "loess",
+                  alpha = 0.2,
+                  aes(fill = algorithm)) +
+      scale_color_manual(name = "Algorithm",
+                         values = c("#e41a1c",
+                                    "#377eb8",
+                                    "#4daf4a",
+                                    "#984ea3",
+                                    "#ff7f00"),
+                         labels = c("pca" = "PCA",
+                                    "ica" = "ICA",
+                                    "nmf" = "NMF",
+                                    "dae" = "DAE",
+                                    "vae" = "VAE")) +
+      scale_fill_manual(name = "Algorithm",
+                        values = c("#e41a1c",
+                                   "#377eb8",
+                                   "#4daf4a",
+                                   "#984ea3",
+                                   "#ff7f00"),
+                        labels = c("pca" = "PCA",
+                                   "ica" = "ICA",
+                                   "nmf" = "NMF",
+                                   "dae" = "DAE",
+                                   "vae" = "VAE")) +
+      theme_bw() +
+      ggtitle(plot_title) +
+      ylab("-log 10 P") +
+      xlab("k Dimensions") +
+      theme(axis.title.x = element_text(size = 7),
+            axis.title.y = element_text(size = 7),
+            axis.text.x = element_text(angle = 90,
+                                     size = 5),
+            axis.text.y = element_text(size = 6),
+            plot.title = element_text(hjust = 0.5,
+                                    size = 8),
+            legend.text = element_text(size = 6),
+            legend.title = element_text(size = 7),
+            legend.key.size = unit(0.7, "lines"))
+}
+
+# Plot
+line_legend <- cowplot::get_legend(line_plot[['Neutrophil']])
+panel_g_gg <- cowplot::plot_grid(
+    line_plot[['Neutrophil']] + theme(legend.position = "none"),
+    line_plot[['Monocyte']] + theme(legend.position = "none"),
+    labels = c("g", ""),
+    ncol = 2
+)
+
+panel_g_gg <- cowplot::plot_grid(
+    panel_g_gg,
+    line_legend,
+    rel_widths = c(1, 0.15),
+    ncol = 2
+)
+
+panel_g_gg
+
+myPalette <- colorRampPalette(rev(c("#43b8d8",
```

Comment:
Not exactly sure what the take-away is for this plot -- just curious.  I'm sure its explained in the paper

Reply:
yeah, i think this is a fun one. I am calling it a career path plot lol

The lines connect continuous k. The colors represent this continuum from blue k = 2 to red k = 200.  I am looking for concordance between the -log10 p value (y axis) for the t-test comparing neutrophil treatments (and monocytes vs. other cell types) and the BioBombe network projection score.

It looks like they are not strongly related, except for NMF models. It also appears that any way you slice it, if you believe in the biobombe z score or the p value, the highest latent space dimension is not always the best at capturing the cell type signatures.